README.md

```diff
@@ -51,6 +51,132 @@ After this, you will have a directory containing files used for creating a
 custom Jupyter widget. You will now be able to easily package and distribute
 your custom Jupyter widget.
 
+## Files Generated by cookiecutter
+Jupyter widgets are made of two parts:
+- Python package for the code executed in kernel as the back-end of the widget
+- Javascript NPM package for the front-end of the widget running in browser
+
+cookiecutter generates source files needed for both back-end and front-end of the widget
+
+```
+├── <python_package_name>
+│   ├── __init__.py
+│   ├── _version.py
+│   └── example.py
+├── js
+│   ├── lib
+│   │   ├── embed.js
+│   │   ├── example.js
+│   │   ├── extension.js
+│   │   ├── index.js
+│   │   └── labplugin.js
+│   ├── package.json
+│   ├── README.md
+│   └── webpack.config.js
+├── .gitignore
+├── <npm_package_name>.json
+├── MANIFEST.in
+├── README.md
+├── RELEASE.md
+├── setup.cfg
+└── setup.py
+```
+
+- **`<python_package_name> directory`**\
+This directory contains the python package that runs in the kernel
+
+    - **`__init__.py`**\
+Marks the `<python_package_name>` directory as a python module, allowing `<python_package_name>` to be imported as a module
+
+    - **`_version.py`**\
+Contains python module version definitions
+
+    - **`example.py`**\
+Widget class definition in python. Defines widget data model that is synced with front-end automatically using traitlets and Backbone.js
+
+- **`js directory`**\
+Source files for widget front-end. Bundled as an npm package\
+
+    - **`lib directory`**\
+Directory containing source files
+
+        - **`embed.js`**\
+Webpack bundler entry point for generating front-end code bundle. The output of this bundle is used for unpkg distribution.
+
+        - **`example.js`**\
+Widget front-end source code. Contains widget model and view class definitions. Widget model is derived from [DOMWidgetModel](https://github.com/jupyter-widgets/ipywidgets/blob/master/packages/base/src/widget.ts) which provides functionality to communicate with the model instance on kernel. Widget view is derived from [DOMWidgetView](https://github.com/jupyter-widgets/ipywidgets/blob/master/packages/base/src/widget.ts) which has a `model` attribute referencing to widget model it is presenting. If a notebook outputs same widget more than once, there is still one model in back-end and one in front-end, only separate views are created.
+
+        - **`extension.js`**\
+Contains the javascript that is run when the notebook is loaded. It contains some requirejs configuration and the `load_ipython_extension` which is required for any notebook extension.
```

Comment:
Maybe we should be consistent with the doc for `embed.js`?

Webpack entry-point for generating the code bundle that is loaded when widgets are used in the classic notebook.